packages/gatsby-transformer-pdf/src/gatsby-node.js

```diff
@@ -0,0 +1,49 @@
+const Promise = require(`bluebird`)
+const PDFParser = require(`pdf2json`)
+
+const convertToJson = path =>
+  new Promise((res, rej) => {
+    const pdfParser = new PDFParser(this, 1)
+    pdfParser.loadPDF(path)
+    pdfParser
+      .on(`pdfParser_dataReady`, pdfData => {
+        res(pdfParser.getRawTextContent())
+      })
+      .on(`pdfParser_dataError`, errData => {
+        rej(`PDF to JSON conversion failed!`)
+      })
+  })
+
+async function onCreateNode({
+  node,
+  actions,
+  loadNodeContent,
+  createNodeId,
+  createContentDigest,
+}) {
+  const { createNode, createParentChildLink } = actions
+
+  // Filter out non-pdf content
+  if (node.extension !== `pdf`) {
+    return
+  }
+
+  let parsedContent = await convertToJson(node.absolutePath)
+
+  const pdfNode = {
+    id: createNodeId(`${node.id} >>> ${node.extension}`),
```

Comment:
Thinking aloud here, how would we best query for a specific pdf document. Something like this?

```graphql
query {
  pdf(relativePath: { regex: "/whatever.pdf/" }) {
    content
  }
}
```

for something like a PDF listing! If you need any help getting this setup with like a starter/example, let me know and we can test it out!

Reply:
You'd query first the File node and then the child PDF node

Reply:
i do something, but i don't know if it is in the right way. I will need some help.